vendor/brave-ios/Ledger/BATBraveLedger.mm

```diff
@@ -190,9 +193,129 @@ - (void)dealloc
   delete rewardsDatabase;
 }
 
+- (void)initializeLedgerService:(BOOL)executeMigrateScript
+{
+  if (self.initialized || self.initializing) {
+    return;
+  }
+  self.initializing = YES;
+  
+  BLOG(ledger::LogLevel::LOG_DEBUG) << "DB: Migrate from CoreData? " << (executeMigrateScript ? "YES" : "NO") << std::endl;
+  ledger->Initialize(executeMigrateScript, ^(ledger::Result result){
+    self.initialized = result == ledger::Result::LEDGER_OK;
+    self.initializing = NO;
+    if (self.initialized) {
+      self.prefs[kMigrationSucceeded] = @(YES);
+      [self savePrefs];
+    } else {
+      if (result == ledger::Result::DATABASE_INIT_FAILED) {
```

Comment:
Is there a chance that the migration fails, but with different error than `DATABASE_INIT_FAILED `?
If so, should we handle such case, or at least log it

Reply:
It _can_ but its unlikely. We never handled it in the past already, but I'll add a log regardless

Reply:
Added a log